src/main/java/com/google/api/codegen/transformer/ruby/RubyGapicSurfaceTransformer.java

```diff
@@ -344,19 +342,17 @@ private ViewModel generateTopLevelIndexView(Model model, GapicProductConfig prod
     RubyPackageMetadataNamer packageNamer =
         new RubyPackageMetadataNamer(productConfig.getPackageName());
 
-    List<String> apiModules = namer.getApiModules();
-    int moduleCount = apiModules.size();
     ImmutableList.Builder<ModuleView> moduleViews = ImmutableList.builder();
 
-    for (int i = 0; i < moduleCount; ++i) {
-      if (i == moduleCount - SERVICE_MODULE_RINDEX) {
+    for (String moduleName : namer.getApiModules()) {
+      if (moduleName.equals(namer.getModuleServiceName())) {
         moduleViews.add(
             metadataTransformer
                 .generateReadmeMetadataView(model, productConfig, packageNamer)
-                .moduleName(apiModules.get(i))
+                .moduleName(moduleName)
                 .build());
-      } else if (i != moduleCount - VERSION_MODULE_RINDEX || includeVersionModule) {
-        moduleViews.add(SimpleModuleView.newBuilder().moduleName(apiModules.get(i)).build());
+      } else if (includeVersionModule || !moduleName.equals(namer.getModuleVersionName())) {
```

Comment:
Nice! This is more readable and seems more robust than checking an index, as well. 👍

Reply:
(Actually, I just realized it does the same thing under the hood in the namer. Still, I like the clearer intent.)